app/controllers/application_controller.rb

```diff
@@ -190,11 +191,24 @@ def expanded_embed(url, caption: nil, link: nil)
       end
     end
 
-    render_to_string partial: "/articles/embeds/#{slug}.html.erb", locals: { embed_id: embed_id || url, caption: caption, link: link }
+    render_to_string partial: "/articles/embeds/#{slug}.html.erb", locals: { embed_id: embed_id || url, caption: caption, link: link, id: id }
   end
   helper_method :expanded_embed
 
   def url_or_path?(string)
     string.match?(/^(http|\/)\S+/)
   end
+
+  def remove_id(pieces)
+    return unless pieces.present?
+
+    id_string = pieces.detect { |piece| piece =~ /id:\S+/ }
+
+    if id_string
+      id = id_string.split(":").last
+      pieces.delete(id_string)
```

Comment:
But I take your point that this could be too confusing.  Would it be less so if I just did this all inline?